pl_examples/basic_examples/conf/trainer_conf.py

```diff
@@ -0,0 +1,56 @@
+from dataclasses import dataclass
+from typing import Union, Optional, List, Dict, Tuple, Iterable, Any
+
+import hydra
+from hydra.core.config_store import ConfigStore
+
+
+@dataclass
+class Lightning_Args:
+    # callbacks: Optional[List[Callback]] = None
```

Comment:
What do callbacks look like in the command line?

Reply:
I have not used them before, this is a todo still for me. I have to experiments with what one looks like and how to use it.

Reply:
I do not think this is possible giving the callback from the command line. I tried a few different ways and nothing functions. I think this comes from a by product of how they generate the cli arguments, which is: https://github.com/PyTorchLightning/pytorch-lightning/blob/f972ab3a828eae1847a793da0b2c25c6074647a4/pytorch_lightning/trainer/trainer.py#L617. Additionally, you can see here how callbacks are defined and passed: https://pytorch-lightning.readthedocs.io/en/latest/callbacks.html.

Reply:
Easy option:
1. User are passing those in as passthrough:
```
hydra.utils.instantiate(**cfg.pl.trainer, callbacks=callbacks)
```

2. Try to use Structured Configs to initialize those.
Flexible lists in general are a bit hard to configure. let's postpone this to a subsequent diff.